### ISSUE TITLE
Helm gives invalid memory address when rendering a ConfigMap template

### ISSUE THREAD

**Issue by CodeJjang** [helm/helm#4965]
I have declared several resources as templates, i.e. `_deployment.yaml`, `_service.yaml`, etc, in a chart named `common`.
Then, it is possible to consume the templates in other charts (requiring the `common` chart) the following manner (i.e. in a regular `deployment.yaml` file):  
```
{{ template "common.deployment" . }}
```
Finally, I have an umbrella chart uniting all the consumers.  

In order to allow the consumers to override the values in the `common` chart, I add these lines in the top of the templated resource definitions:
```
{{- define "common.<resource_name>" -}}
{{- $common := dict "Values" .Values.common -}} 
{{- $noCommon := omit .Values "common" -}} 
{{- $overrides := dict "Values" $noCommon -}} 
{{- $noValues := omit . "Values" -}} 
{{- with merge $noValues $overrides $common -}}
...
```

This works great with resources such as `Deployment`, `Service`, `Ingress` etc, however it fails when I try to implement this with a `ConfigMap`:
```
Error: rendering template failed: runtime error: invalid memory address or nil pointer dereference
```

I've uploaded the code to a [public repo](https://github.com/CodeJjang/helm-microservices-example) to make it easier to reproduce.

1. If you run:
```
./helm-dep-up-umbrella.sh helm-chart/  && helm install helm-chart/ --dry-run
```
it works.  
(The first script is used to update the consumers dependencies and then the umbrella chart dependencies since --dep-up is not recursive).
2. Now, replace `_configmap.yaml` file in `common/templates/` with the following content:
```
{{- define "common.configmap" -}}
{{- $common := dict "Values" .Values.common -}} 
{{- $noCommon := omit .Values "common" -}} 
{{- $overrides := dict "Values" $noCommon -}} 
{{- $noValues := omit . "Values" -}} 
{{- with merge $noValues $overrides $common -}}
apiVersion: v1
kind: ConfigMap
{{ template "common.metadata" . }}
data:
  .env:
{{ .Files.Get "dotenv" | indent 4 }}
{{- end -}}
{{- end -}}
```
3. Run comman from step 1.
4. Error reproduced.  

I've noticed that if I remove `$common` from the `merge` function then everything works (but the expected overriding behaviour won't occur), so it might be something with the order of how everything is rendered.

Would love your help and insights regarding this issue.  

Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"clean", BuildDate:"2018-10-24T06:54:59Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:44:10Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
Minikube, also tested on private K8S cluster

**Comment by kierenj** [comment]
I'm getting the same when trying to do something similar (combining dicts to override values).

`lint` works fine, but `helm install --debug --dry-run` gives `Error: rendering template failed: runtime error: invalid memory address or nil pointer dereference`.

`Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}`
`Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}`

**Comment by k20human** [comment]
Hi,

Same problem as @CodeJjang with ConfigMap.
Did you find a solution ?

**Comment by timothydowney** [comment]
I'm seeing same when doing something like this:

```
{{- $masterValues := (merge (dict "Values" .Values.master) .) }}
```

I'm basically try to deal with a situation using common templates where I take sections of the values and push them to the top level to pass into the templates.  What is odd is that it stopped working only after using it multiple times when rendering an overall chart (i.e. i used it to build the StatefulSet and a Service, but now it blows up on the Ingress).

In other words, the error appears only after I do this more than twice in the same overall chart lint.

**Comment by bacongobbler** [comment]
closing as inactive.
